src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchesSpec.groovy

```diff
@@ -333,4 +335,90 @@ class SwitchesSpec extends HealthCheckSpecification {
                  operation: { getNorthbound().validateSwitch(NON_EXISTENT_SWITCH_ID) }]
         ]
     }
+
+    @Tidy
+    @Tags(LOW_PRIORITY)
+    @Unroll
+    def "Able to partially update switch a 'location.#data.field' field"() {
+        given: "A switch"
+        def sw = topology.activeSwitches.first()
+        def initConf = northbound.getSwitch(sw.dpId)
+
+        when: "Request a switch partial update for a #data.field field"
+        def updateRequest = new SwitchPatchDto().tap {
+            it.location = new SwitchLocationDtoV2().tap { it."$data.field" = data.newValue }
+        }
+        def response = northboundV2.partialSwitchUpdate(sw.dpId, updateRequest)
+
+        then: "Update response reflects the changes"
+        response.location."$data.field" == data.newValue
+
+        and: "Changes actually took place"
+        northbound.getSwitch(sw.dpId).location."$data.field" == data.newValue
+
+        cleanup:
```

Comment:
Just create a method that maps SwitchDto (initConf) to SwitchPatchDto and get rid of all these if-else branches

Reply:
I didn't extract it into separate method(I will do it when we start use this kind of operation more often).
For now I just refactored this block of code